### PR TITLE
Minor improvements to detekt-rules

### PR DESCRIPTION
**Pull request by 3flex**
I was browsing through the rules and spotted a few places where I felt the code and wording could be tidied up slightly, or made clearer.

These are all very minor changes, but broke out into multiple commits only to ease the review.

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt [hunk 1]
```diff
@@ -70,12 +69,12 @@ class UnusedPrivateMember(config: Config = Config.empty) : Rule(config) {
 		}
 	}
 
-	override fun visitClassOrObject(classOrObject: KtClassOrObject) {
-		if ((classOrObject as? KtClass)?.isInterface() == true) {
+	override fun visitClass(klass: KtClass) {
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt]: I think this change is not right?
Interfaces should get skipped but not objects ^^

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/MultiRuleCollector.kt [hunk 2]
```diff
@@ -89,7 +89,7 @@ class MultiRuleVisitor : DetektVisitor() {
 			return
 		}
 
-		if (property.isOverridden() && property.name != null && property.name == "rules") {
+		if (property.isOverride() && property.name != null && property.name == "rules") {
```
Comment (schalkms) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/MultiRuleCollector.kt]: Sounds weird. I can live with it though.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt (hunk 1 above)
Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt]: I second that.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/EqualsWithHashCodeExist.kt [hunk 3]
```diff
@@ -71,15 +69,14 @@ class EqualsWithHashCodeExist(config: Config = Config.empty) : Rule(config) {
 		super.visitFile(file)
 	}
 
-	override fun visitClassOrObject(classOrObject: KtClassOrObject) {
-		val klass = classOrObject as? KtClass
-		if (klass != null && klass.isDataClass()) {
+	override fun visitClass(klass: KtClass) {
```
Comment (schalkms) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/EqualsWithHashCodeExist.kt]: A Kotlin `Object` can also have an `equals` and `hashCode` method. Please revert this.

**Pull request merged**